ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -311,11 +360,17 @@ private void readWithProgress(
       int requestId,
       ResponseBody responseBody) throws IOException {
     Reader reader = responseBody.charStream();
+    ProgressResponseBody progressResponseBody = (ProgressResponseBody)responseBody;
```

Comment:
I see. So is `addNetworkInterceptor` never called/used?

In any case, we can make `readWithProgress` be more resilient against that case for the time being. We wouldn't lose any functionality we don't currently have, just that FB internally would not get useful information with the `'progress'` event (which used to not be dispatched anyway).

How about something like this:

```java
  private void readWithProgress(
      RCTDeviceEventEmitter eventEmitter,
      int requestId,
      ResponseBody responseBody) throws IOException {
    Reader reader = responseBody.charStream();
    int totalBytesRead = -1;
    int contentLength = -1;
    try {
      ProgressResponseBody progressResponseBody = (ProgressResponseBody)responseBody;
      totalBytesRead = progressResponseBody.totalBytesRead();
      contentLength = progressResponseBody.contentLength());
    } catch (ClassCastException e) {
      // Ignore
    }

    try {
      char[] buffer = new char[MAX_CHUNK_SIZE_BETWEEN_FLUSHES];
      int read;
      while ((read = reader.read(buffer)) != -1) {
        ResponseUtil.onIncrementalDataReceived(
          eventEmitter,
          requestId,
          new String(buffer, 0, read),
          totalBytesRead,
          contentLength
      }
    } finally {
      reader.close();
    }
  }
```